### PR TITLE
Delete `behavior/` tests + `codegen.*pp` snippets test

### PR DESCRIPTION
**Pull request by vurusovs**
### Tickets:
 - [CVS-130307](https://jira.devtools.intel.com/browse/CVS-130307)
 - [CVS-130342](https://jira.devtools.intel.com/browse/CVS-130342)

### CODE REVIEW

**Comment by ilya-lavrenov** [comment]
are they all supposed to be covered by 2.0 analogues?

**Review by iefode — APPROVED** [review]
LGTM

**Review by ilya-lavrenov — APPROVED** [review]

**Review by IvanNovoselov — APPROVED** [review]

**Review by zhaixuejun1993 — APPROVED** [review]
LGTM for AUTO BATCH Plugin.

**Comment by ilya-lavrenov** [comment]
Please, fix compilation on Windows CC

File: src/tests/functional/plugin/shared/include/behavior/ov_plugin/auto_batching_tests.hpp [hunk 1]
```diff
@@ -86,7 +86,7 @@ class AutoBatching_Test : public OVPluginTestBase,
             for (size_t j = 0; j < num_requests; j++) {
                 outputs.push_back(output);
                 outElementsCount.push_back(
-                        std::accumulate(begin(fn_ptrs[i]->get_output_shape(0)), end(fn_ptrs[i]->get_output_shape(0)), 1,
+                        std::accumulate(begin(fn_ptrs[i]->get_output_shape(0)), end(fn_ptrs[i]->get_output_shape(0)), size_t(1),
```
Comment (ilya-lavrenov) [inline src/tests/functional/plugin/shared/include/behavior/ov_plugin/auto_batching_tests.hpp:89]: please, fix a compilation error introduced in current PR
See https://github.com/openvinotoolkit/openvino/actions/runs/7672375947/job/20912751386?pr=22469#step:15:6782

**Pull request merged**